src/fvm_multicell.hpp

```diff
@@ -265,19 +275,33 @@ class fvm_multicell {
     };
 
     /// events staged for upcoming integration stage
-    std::vector<deliverable_event> staged_events_;
+    std::vector<std::vector<deliverable_event>> staged_events_;
 
     /// event queue for integration period
     /// TODO: Placeholder; this will change when event lists are moved to
     /// a backend structure.
-    event_queue<deliverable_event> events_;
+    using cell_event_queue = multi_event_stream<deliverable_event>;
+    cell_event_queue events_;
+
+    bool no_pending_events() const {
```

Comment:
I would rename this `has_pending_events()` and use `!events_.has_pending_events()`.

Because I prefer `true` to indicate presence, and `false` to indicate absence (I am easily confused).

Discuss.